Web Inspector: Use nesting for the variables inside :root in Variables.css

francescorn opened the pull request:
#### fe05198178024cccbf058d1759ffb64681708413
<pre>
Web Inspector: Use nesting for the variables inside :root in Variables.css
<a href="https://bugs.webkit.org/show_bug.cgi?id=268940">https://bugs.webkit.org/show_bug.cgi?id=268940</a>

Reviewed by Tim Nguyen and Matthieu Dubet.

Nest all elements inside of :root and body property in Variables.css

Co-authored-by: Tim Nguyen &lt;nt1m@users.noreply.github.com&gt;

* Source/WebInspectorUI/UserInterface/Views/Variables.css:
(:root):
(body):
(&amp;:is(.mac-platform.monterey, .mac-platform.big-sur)):
(&amp;:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)):
(&amp;:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)):
(&amp;.mac-platform):
(body.window-inactive): Deleted.
(body:is(.mac-platform.monterey, .mac-platform.big-sur)): Deleted.
(body:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)): Deleted.
(body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)): Deleted.
(@media (prefers-color-scheme: dark) :root): Deleted.
(@media (prefers-color-scheme: dark) body.window-inactive): Deleted.
(@media (prefers-color-scheme: dark) body:is(.mac-platform.monterey, .mac-platform.big-sur)): Deleted.
(body.mac-platform): Deleted.

Canonical link: <a href="https://commits.webkit.org/274987@main">https://commits.webkit.org/274987@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a9d500db561ec5a0f80dfff99b625ebff25a6e9e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40591 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19604 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42969 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43144 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36680 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22565 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16935 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/33678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16543 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/34995 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14261 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/35968 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44417 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/36296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40032 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15406 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12631 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38363 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17025 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9091 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17076 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16669 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -23,7 +23,7 @@
  */
 
-:root {
+body {
```
Variables can stay on `:root`, I don't see a reason to move them to `body`

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -408,7 +407,6 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
         --slider-track-tick-background: hsl(0, 0%, 55%);
     }
```
The @media should be nested inside the :root rule, otherwise this won't work.

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -31,40 +30,27 @@
-
     --background-color-content: white;
     --background-color-intermediate: hsl(0, 0%, 97%);
-
```
I would undo the newline changes.

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -354,100 +240,115 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
         --slider-track-tick-background: hsl(0, 0%, 55%);
     }
+}
+body {
```
```suggestion
}

body {
```
nit: add a new line for readability.

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -354,100 +240,115 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
+        --glyph-color-disabled: hsla(0, 0%, 30%, 0.35);
+        --glyph-opacity: 0.6;
+        --border-color: hsl(0, 0%, 85%);
+        @media (prefers-color-scheme: dark) {
```
```suggestion
        --border-color: hsl(0, 0%, 85%);

        @media (prefers-color-scheme: dark) {
```
nit: newline for readability

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -354,100 +240,115 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
+            --glyph-color-disabled: hsla(0, 0%, 52%, 0.4);
+        }
+    }
+    &:is(.mac-platform.monterey, .mac-platform.big-sur) {
```
```suggestion
    }

    &:is(.mac-platform.monterey, .mac-platform.big-sur) {
```
nit: newline for readability

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -354,100 +240,115 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
-        --text-color: hsl(0, 0%, 85%);
-        --text-color-transparent: hsla(0, 0%, 85%, 0.4);
+        --sorted-header-font-weight: 700;
+        @media (prefers-color-scheme: dark) {
```
```suggestion
        --sorted-header-font-weight: 700;

        @media (prefers-color-scheme: dark) {
```
nit: newline for readability

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -354,100 +240,115 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
+        /* keep in sync with `WI.undockedTitleAreaHeight` */
+        --undocked-title-area-height: calc(27px / var(--zoom-factor));
+    }
+    &.mac-platform {
```
ditto for the three rules here
```
     }

     &:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
         /* keep in sync with `WI.undockedTitleAreaHeight` */
         --undocked-title-area-height: calc(22px / var(--zoom-factor));
     }

     &:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
         /* keep in sync with `WI.undockedTitleAreaHeight` */
         --undocked-title-area-height: calc(27px / var(--zoom-factor));
     }

     &.mac-platform {
 ```

mdubet reviewed (approved):
Not sure about the line break changes, but otherwise it's a good example of how nesting improves CSS readability 👍

nt1m commented on Source/WebInspectorUI/UserInterface/Views/Variables.css:
```diff
@@ -407,47 +366,80 @@ body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
+        /* keep in sync with `WI.undockedTitleAreaHeight` */
+        --undocked-title-area-height: calc(27px / var(--zoom-factor));
+    }
+    &.mac-platform {
```
```suggestion
    }

    &:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
        /* keep in sync with `WI.undockedTitleAreaHeight` */
        --undocked-title-area-height: calc(22px / var(--zoom-factor));
    }

    &:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked) {
        /* keep in sync with `WI.undockedTitleAreaHeight` */
        --undocked-title-area-height: calc(27px / var(--zoom-factor));
    }

    &.mac-platform {
```

francescorn committed:
Web Inspector: Use nesting for the variables inside :root in Variables.css
https://bugs.webkit.org/show_bug.cgi?id=268940

Reviewed by Tim Nguyen and Matthieu Dubet.

Nest all elements inside of :root and body property in Variables.css

Co-authored-by: Tim Nguyen <nt1m@users.noreply.github.com>

* Source/WebInspectorUI/UserInterface/Views/Variables.css:
(:root):
(body):
(&:is(.mac-platform.monterey, .mac-platform.big-sur)):
(&:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)):
(&:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)):
(&.mac-platform):
(body.window-inactive): Deleted.
(body:is(.mac-platform.monterey, .mac-platform.big-sur)): Deleted.
(body:not(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)): Deleted.
(body:is(.mac-platform.monterey, .mac-platform.big-sur):not(.docked)): Deleted.
(@media (prefers-color-scheme: dark) :root): Deleted.
(@media (prefers-color-scheme: dark) body.window-inactive): Deleted.
(@media (prefers-color-scheme: dark) body:is(.mac-platform.monterey, .mac-platform.big-sur)): Deleted.
(body.mac-platform): Deleted.

Canonical link: https://commits.webkit.org/274987@main
